---
title: "ImmuneMicrobiomeNew"
author: "Yuka Moroishi"
date: "March 4, 2019"
output: html_document
---

#Loading Data
```{r}
#Loading Questionnaire
load("/afs/northstar.dartmouth.edu/udata/margk/collab/AnneHoen/Lab/YukaMoroishi/SourceFiles/NHBCSreq20dec2017ah_v21Feb2018/req01102018ah_21feb2018.rdata")

#Loading Key
asvkey = read.csv("/Volumes/f00359t/Dissertation Materials/CordProject/mbl.6W.12M.st.pair.csv",header = FALSE)

#Loading Immune Cell Proportions
immune.original = read.csv("/afs/northstar.dartmouth.edu/udata/margk/collab/AnneHoen/Lab/YukaMoroishi/SourceFiles/NHBCSreq20mar2018ym/NHBCSreq20March2018ym-deided.csv")

#Loading Microbiome Data
setwd("/afs/northstar.dartmouth.edu/udata/margk/collab/AnneHoen/Lab/YukaMoroishi/SourceFiles/JCM_COBR/dada2/sv6W")

#loading New immune cell data 
library(openxlsx)
immune.new = read.xlsx("//afs/northstar.dartmouth.edu/udata/margk/collab/AnneHoen/Lab/YukaMoroishi/SourceFiles/NHBCSreq20mar2018ym/NHBCSreq19SEP2018ym-deided.xlsx")

#Loading Microbiome Data
setwd("/afs/northstar.dartmouth.edu/udata/margk/collab/AnneHoen/Lab/YukaMoroishi/SourceFiles/JCM_COBR/dada2/sv6W")
files = list.files()
files = files[grep(files, pattern = "rds")]
mydata = lapply(files, readRDS)
names(mydata) = files
asvdata.original = mydata[["seqtab.nochim.rds"]]
#Column names of ASV 
asv.colnames = mydata[["seqtab.nochim.colnames.rds"]]
#rm(mydata)
#Genus Level
genusdata.original = mydata[["RDPtaxtab_G_6W.rds"]]

genus.colnames = colnames(genusdata.original)
```

#Data Clean Up
```{r}
immune = immune.original
immune[,17] = as.factor(immune[,17])
asvdata = asvdata.original
genusdata = genusdata.original

immune$mblid = as.character(immune$mblid)
immune$mblid = gsub(" ", "", immune$mblid, fixed = TRUE)
matches = match(immune$mblid, row.names(asvdata))
matches = na.omit(matches)
asvdata = asvdata[matches,]
genusdata = genusdata[matches,]

matches2 = match(row.names(asvdata),immune$mblid)
matches2 = na.omit(matches2)
immune = immune[matches2,]
immune.new2 = immune.new[matches2,]


immune.median = immune[,10:17]
immune.median[,1:7] = apply(immune[,10:16], 2, function(x) ifelse(x>median(x), 1, 0))


immune.median2 = immune.new2[,18:24]
immune.median22 = apply(immune.median2, 2, function(x) ifelse(x>median(x), 1, 0))
```



```{r}
#Phygenetic Tree

source("https://bioconductor.org/biocLite.R")
biocLite("DECIPHER")
biocLite("phangorn")
biocLite("phytools")
biocLite("ips")
biocLite("dada2")
library("DECIPHER")
library("phangorn")
library("phytools")
library("ips")
library("dada2")

####Generating a new phylogenetic tree using fasttree - this works much better ####
#source of assistance: @gitares https://github.com/benjjneb/dada2/issues/88
#http://web.stanford.edu/class/bios221/MicrobiomeWorkflowII.html

asvdata2 = asvdata
colnames(asvdata2) = asv.colnames
seqs.fasttree <- getSequences(asvdata2)
#names(seqs.fasttree) <- asv.colnames
alignment.fasttree <- AlignSeqs(DNAStringSet(seqs.fasttree), anchor = NA, verbose = TRUE)
phagAlign.fasttree <- phyDat(as(alignment.fasttree, "matrix"), type="DNA")
write.phyDat(phagAlign.fasttree, file = "alignment-fasttree.fasta", format = "fasta")

#start terminal and navigate to working directory (which should contain the fasttree program)
#gcc -lm -O2 -Wall -o FastTree FastTree.c
#./FastTree -gtr -nt alignment-fasttree.fasta > fasttree.tre
 
#rooting the tree after using the alignment-fasttree.fasta file to generate a tree from fast tree (tree unrooted)
fasttree.tree <- read.tree("fasttree.tre") #this tree is unrooted
is.rooted(fasttree.tree)
tree.rooted <- midpoint.root(fasttree.tree) #midpoint root from phytools
plot(tree.rooted)
is.rooted(tree.rooted)

library(GUniFrac)
require(ade4)
colnames(asvdata2) = as.character(seq(from = 1, to = ncol(asvdata2), by = 1))
asvrare = Rarefy(asvdata2)$otu.tab.rff
unifracresults = GUniFrac(asvrare, tree.rooted, alpha=c(0, 0.5, 1))$unifracs
dd5 <- unifracresults[, , "d_0.5"]

#ASV Level
#bcell
adonis(as.dist(dd5) ~ immune.median22[,"bcell_revised"])
s.class(cmdscale(dd5, k=2), 
        fac = as.factor(immune.median22[,"bcell_revised"]), 
        col = c("red","blue"),label=invisible())

#cd4t
adonis(as.dist(dd5) ~ immune.median22[,"cd4t_revised"])
s.class(cmdscale(dd5, k=2), fac = as.factor(immune.median22[,"cd4t_revised"]), col = c("red","blue"),label=invisible())

#cd8t
adonis(as.dist(dd5) ~ immune.median22[,"cd8t_revised"])
s.class(cmdscale(dd5, k=2), fac = as.factor(immune.median22[,"cd8t_revised"]), col = c("red","blue"),label=invisible())

#gran
adonis(as.dist(dd5) ~ immune.median22[,"gran_revised"])
s.class(cmdscale(dd5, k=2), fac = as.factor(immune.median22[,"gran_revised"]), col = c("red","blue"),label=invisible())

#mono 
adonis(as.dist(dd5) ~ immune.median22[,"mono_revised"])
s.class(cmdscale(dd5, k=2), fac = as.factor(immune.median22[,"mono_revised"]), col = c("red","blue"),label=invisible())

#nk
adonis(as.dist(dd5) ~ immune.median22[,"nk_revised"])
s.class(cmdscale(dd5, k=2), fac = as.factor(immune.median22[,"nk_revised"]), col = c("red","blue"),label=invisible())

#nrbc
adonis(as.dist(dd5) ~ immune.median22[,"nrbc_revised"])
s.class(cmdscale(dd5, k=2), fac = as.factor(immune.median22[,"nrbc_revised"]), col = c("red","blue"),label=invisible())


```


```{r}
#Using Bray-Curtis
library(vegan)
asvrarebc = Rarefy(asvdata2)$otu.tab.rff
genusrarebc = Rarefy(genusdata)$otu.tab.rff

asvdata2bc = vegdist(asvrarebc, method="bray")
genusdata2bc = vegdist(genusrarebc, method="bray")
asv.bc.clust = hclust(asvdata2bc, method = "complete")

#bcell
adonis(asvdata2bc ~ immune.median22[,"bcell_revised"])
s.class(cmdscale(asvdata2bc, k=2), fac = as.factor(immune.median22[,"bcell_revised"]), col = c("red","blue"),label=invisible())

#cd4t - Marginally. P = 0.106
adonis(asvdata2bc ~ immune.median22[,"cd4t_revised"])
s.class(cmdscale(asvdata2bc, k=2), fac = as.factor(immune.median22[,"cd4t_revised"]), col = c("red","blue"),label=invisible())

#cd8t
adonis(asvdata2bc ~ immune.median22[,"cd8t_revised"])
s.class(cmdscale(asvdata2bc, k=2), fac = as.factor(immune.median22[,"cd8t_revised"]), col = c("red","blue"),label=invisible())

#gran
adonis(asvdata2bc ~ immune.median22[,"gran_revised"])
s.class(cmdscale(asvdata2bc, k=2), fac = as.factor(immune.median22[,"gran_revised"]), col = c("red","blue"),label=invisible())

#mono
adonis(asvdata2bc ~ immune.median22[,"mono_revised"])
s.class(cmdscale(asvdata2bc, k=2), fac = as.factor(immune.median22[,"mono_revised"]), col = c("red","blue"),label=invisible())

#nk - P = 0.149
adonis(asvdata2bc ~ immune.median22[,"nk_revised"])
s.class(cmdscale(asvdata2bc, k=2), fac = as.factor(immune.median22[,"nk_revised"]), col = c("red","blue"),label=invisible())

#nrbc
adonis(asvdata2bc ~ immune.median22[,"nrbc_revised"])
s.class(cmdscale(asvdata2bc, k=2), fac = as.factor(immune.median22[,"nrbc_revised"]), col = c("red","blue"),label=invisible())


```

#Genus level
```{r}
genusrarebc = Rarefy(genusdata)$otu.tab.rff
genusdata2bc = vegdist(genusrarebc, method="bray")


#bcell
adonis(genusdata2bc ~ immune.median22[,"bcell_revised"])
s.class(cmdscale(genusdata2bc, k=2), fac = as.factor(immune.median22[,"bcell_revised"]), col = c("red","blue"),label=invisible())

#cd4t - Not significant at genus level
adonis(genusdata2bc ~ immune.median22[,"cd4t_revised"])
s.class(cmdscale(genusdata2bc, k=2), fac = as.factor(immune.median22[,"cd4t_revised"]), col = c("red","blue"),label=invisible())

#cd8t
adonis(genusdata2bc ~ immune.median22[,"cd8t_revised"])
s.class(cmdscale(genusdata2bc, k=2), fac = as.factor(immune.median22[,"cd8t_revised"]), col = c("red","blue"),label=invisible())

#gran
adonis(genusdata2bc ~ immune.median22[,"gran_revised"])
s.class(cmdscale(genusdata2bc, k=2), fac = as.factor(immune.median22[,"gran_revised"]), col = c("red","blue"),label=invisible())

#mono
adonis(genusdata2bc ~ immune.median22[,"mono_revised"])
s.class(cmdscale(genusdata2bc, k=2), fac = as.factor(immune.median22[,"mono_revised"]), col = c("red","blue"),label=invisible())

#nk - P = 0.149
adonis(genusdata2bc ~ immune.median22[,"nk_revised"])
s.class(cmdscale(genusdata2bc, k=2), fac = as.factor(immune.median22[,"nk_revised"]), col = c("red","blue"),label=invisible())

#nrbc
adonis(genusdata2bc ~ immune.median22[,"nrbc_revised"])
s.class(cmdscale(genusdata2bc, k=2), fac = as.factor(immune.median22[,"nrbc_revised"]), col = c("red","blue"),label=invisible())


```
#Alpha diversity
```{r}
H <- diversity(asvdata)
simp <- diversity(asvdata, "simpson")
invsimp <- diversity(asvdata, "inv")
## Unbiased Simpson (Hurlbert 1971, eq. 5) with rarefy:
unbias.simp <- rarefy(asvdata, 2) - 1
## Fisher alpha
alpha <- fisher.alpha(asvdata)
## Plot all
pairs(cbind(H, simp, invsimp, unbias.simp, alpha), pch="+", col="blue")

#Shannon Index
t.test(diversity(asvdata)~immune.median22[,"bcell_revised"])
t.test(diversity(asvdata)~immune.median22[,"cd4t_revised"])
t.test(diversity(asvdata)~immune.median22[,"cd8t_revised"])
t.test(diversity(asvdata)~immune.median22[,"mono_revised"])
t.test(diversity(asvdata)~immune.median22[,"gran_revised"])
t.test(diversity(asvdata)~immune.median22[,"nk_revised"])
t.test(diversity(asvdata)~immune.median22[,"nrbc_revised"])

par(mfrow=c(2,4))
boxplot(diversity(asvdata)~immune.median22[,"bcell_revised"], 
        ylab = "H", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "Bcell")


boxplot(diversity(asvdata)~immune.median22[,"cd4t_revised"], 
        ylab = "H", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "CD4T")

boxplot(diversity(asvdata)~immune.median22[,"cd8t_revised"], 
        ylab = "H", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "CD8T")

boxplot(diversity(asvdata)~immune.median22[,"mono_revised"], 
        ylab = "H", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "Monocyte")

boxplot(diversity(asvdata)~immune.median22[,"gran_revised"], 
        ylab = "H", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "Granulocyte")

boxplot(diversity(asvdata)~immune.median22[,"nk_revised"], 
        ylab = "H", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "Natural Killer")

boxplot(diversity(asvdata)~immune.median22[,"nrbc_revised"], 
        ylab = "H", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "nRBC")


#Simpsom Index
t.test(diversity(asvdata, "simpson")~immune.median22[,"bcell_revised"])
t.test(diversity(asvdata, "simpson")~immune.median22[,"cd4t_revised"])
t.test(diversity(asvdata, "simpson")~immune.median22[,"cd8t_revised"])
t.test(diversity(asvdata, "simpson")~immune.median22[,"mono_revised"])
t.test(diversity(asvdata, "simpson")~immune.median22[,"gran_revised"])
t.test(diversity(asvdata, "simpson")~immune.median22[,"nk_revised"])
t.test(diversity(asvdata, "simpson")~immune.median22[,"nrbc_revised"])

par(mfrow=c(2,4))
boxplot(diversity(asvdata, "simpson")~immune.median22[,"bcell_revised"], 
        ylab = "D", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "Bcell")


boxplot(diversity(asvdata, "simpson")~immune.median22[,"cd4t_revised"], 
        ylab = "D", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "CD4T")

boxplot(diversity(asvdata, "simpson")~immune.median22[,"cd8t_revised"], 
        ylab = "D", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "CD8T")

boxplot(diversity(asvdata, "simpson")~immune.median22[,"mono_revised"], 
        ylab = "D", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "Monocyte")

boxplot(diversity(asvdata, "simpson")~immune.median22[,"gran_revised"], 
        ylab = "D", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "Granulocyte")

boxplot(diversity(asvdata, "simpson")~immune.median22[,"nk_revised"], 
        ylab = "D", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "Natural Killer")

boxplot(diversity(asvdata, "simpson")~immune.median22[,"nrbc_revised"], 
        ylab = "D", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "nRBC")
```


#12M
```{r}
#Loading Microbiome Data
setwd("/afs/northstar.dartmouth.edu/udata/margk/collab/AnneHoen/Lab/YukaMoroishi/SourceFiles/JCM_COBR/dada2/sv12M")

#loading New immune cell data 

files = list.files()
files = files[grep(files, pattern = "rds")]
mydata = lapply(files, readRDS)
names(mydata) = files
asvdata.original12 = mydata[["seqtab.nochim.rds"]]
#Column names of ASV 
asv.colnames12 = mydata[["seqtab.nochim.colnames.rds"]]
#rm(mydata)
#Genus Level
genusdata.original12 = mydata[["RDPtaxtab_G_12M.rds"]]

genus.colnames12 = colnames(genusdata.original12)

asvdata12M = asvdata.original12
genusdata12M = genusdata.original12

#immune$mblid = as.character(immune$mblid)
#immune$mblid = gsub(" ", "", immune$mblid, fixed = TRUE)
immune.new$mblid = gsub("    ", "", immune.new$mblid, fixed = TRUE)

#mblid2match = match(immune.new$mblid, asvkey$V2)
#mblid2 = asvkey$V3[mblid2match]


matches12M = match(immune.new$mblid, row.names(asvdata12M))
matches12M = na.omit(matches12M)
asvdata12M = asvdata12M[matches12M,]
genusdata12M = genusdata12M[matches12M,]

matches212M = match(row.names(asvdata12M),immune.new$mblid)
matches212M = na.omit(matches212M)
immune.new212M = immune.new[matches212M,]


asvkey$V2[na.omit(match(row.names(asvdata12M), asvkey$V3))]
length(na.omit(match(row.names(asvdata), asvkey$V2[na.omit(match(row.names(asvdata12M), asvkey$V3))])))
```

#Correlation
```{r}
library(ggcorrplot)

cor.im6 = cor(cbind(immune.new2[,18:24], genusdata.nozeros))
row.names(cor.im6) = c("bcell", "cd4t", "cd8t", "gran", "mono", "nk", 
                       "nrbc",
                       as.character(seq(from = 1, 
                                        to = ncol(genusdata.nozeros),
                                        by = 1)))

colnames(cor.im6) = row.names(cor.im6)

ggcorrplot(cor.im6[8:nrow(cor.im6),1:7])
```

#Proportions
```{r}
df.cd4t = as.data.frame(immune.new2[,"cd4t_revised"])
colnames(df.cd4t) = c("cd4t")

genusdata.nozeros = genusdata[,-which(colSums(genusdata) == 0)]
genusdata.nozeros.prop = prop.table(as.matrix(genusdata.nozeros), margin=1)


library(dplyr)
genusdata.nozeros.prop.01 = ifelse(genusdata.nozeros.prop>0,1,0)
genusdata.nozeros.prop.colsum = colSums(genusdata.nozeros.prop.01)
hist(genusdata.nozeros.prop.colsum)

genusdata.nozeros.prop.2 = genusdata.nozeros.prop
genusdata.nozeros.prop.2 = as.data.frame(genusdata.nozeros.prop.2)
```


```{r}
#Creating files
df.cd4t = as.data.frame(immune.new2[,"cd4t_revised"])
colnames(df.cd4t) = c("cd4t")
df.cd8t = as.data.frame(immune.new2[,"cd8t_revised"])
colnames(df.cd8t) = c("cd8t")
df.bcell = as.data.frame(immune.new2[,"bcell_revised"])
colnames(df.bcell) = c("bcell")
df.gran = as.data.frame(immune.new2[,"gran_revised"])
colnames(df.gran) = c("gran")
df.mono = as.data.frame(immune.new2[,"mono_revised"])
colnames(df.mono) = c("mono")
df.nk = as.data.frame(immune.new2[,"nk_revised"])
colnames(df.nk) = c("nk")
df.nrbc = as.data.frame(immune.new2[,"nrbc_revised"])
colnames(df.nrbc) = c("nrbc")

```



#Adjusted Analysis
```{r}
#Imputing missing data for ever cig
library(mice)
dataimp = as.data.frame(immune.new2[,c("cd4t_revised", "cd8t_revised", "bcell_revised", "gran_revised", "mono_revised", "nk_revised", "nrbc_revised", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")])
tempData <- mice(dataimp,m=5,maxit=50,meth='pmm',seed=500)
completedData <- complete(tempData,1)





#Creating files for adjusted analyses
df.cd4tadj = as.data.frame(completedData[,c("cd4t_revised", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")])
colnames(df.cd4tadj) = c("cd4t", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")
df.cd8tadj = as.data.frame(completedData[,c("cd8t_revised", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")])
colnames(df.cd8tadj) = c("cd8t", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")
df.bcelladj = as.data.frame(completedData[,c("bcell_revised", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")])
colnames(df.bcelladj) = c("bcell", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")
df.granadj = as.data.frame(completedData[,c("gran_revised", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")])
colnames(df.granadj) = c("gran", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")
df.monoadj = as.data.frame(completedData[,c("mono_revised", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")])
colnames(df.monoadj) = c("mono", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")
df.nkadj = as.data.frame(completedData[,c("nk_revised", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")])
colnames(df.nkadj) = c("nk", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")
df.nrbcadj = as.data.frame(completedData[,c("nrbc_revised", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")])
colnames(df.nrbcadj) = c("nrbc", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")



save(df.bcelladj, genusdata.nozeros.prop.2, genusdata.nozeros, df.cd4tadj, df.cd8tadj, df.granadj, df.monoadj, df.nkadj, df.nrbcadj,df.cd8tcd4tadj, file = "inputdataMZILN.Rdata")

```




```{r}
stool_breastmilk_16s = read.csv("/afs/northstar.dartmouth.edu/users/m/margk.collab/AnneHoen/Lab/DupeCoker/GroupFiles/SourceFiles/NHBCSreq03Jun2020_MCocker/stool_breastmilk_16s.csv")

rownames(asvdata2)

stool_breastmilk_16s_73 = stool_breastmilk_16s[na.omit(match(rownames(asvdata2), stool_breastmilk_16s$MBL_ID)),]
stool_breastmilk_16s_73$unq_id
stool_breastmilk_16s_73$MBL_ID

metadata_aim2 = read.csv("/afs/northstar.dartmouth.edu/udata/margk/collab/YukaMoroishi/SourceFiles/NHBCSreq19jul2019_YukaMoroishi/req19jul2019_ym_06jul2020.csv", na.strings=c("", "NA"))

metadata_aim2_73 = metadata_aim2[na.omit(match(stool_breastmilk_16s_73$unq_id, metadata_aim2$unq_id)),]

table(metadata_aim2_73$deliverytype)

summary(glm(immune.new2$bcell_revised ~ metadata_aim2_73$deliverytype))
summary(glm(immune.new2$mono_revised ~ metadata_aim2_73$deliverytype))
summary(glm(immune.new2$cd4t_revised ~ factor(metadata_aim2_73$deliverytype))) #Sig
summary(glm(immune.new2$cd8t_revised ~ metadata_aim2_73$deliverytype))
summary(glm(immune.new2$gran_revised ~ metadata_aim2_73$deliverytype))
summary(glm(immune.new2$nk_revised ~ metadata_aim2_73$deliverytype))
summary(glm(immune.new2$nrbc_revised ~ metadata_aim2_73$deliverytype))
```

