---
title: "ImmuneMicrobiomeNew"
author: "Yuka Moroishi"
date: "March 4, 2019"
output: html_document
---

#Loading Data
```{r}
#Loading Questionnaire
load("/afs/northstar.dartmouth.edu/udata/margk/collab/AnneHoen/Lab/YukaMoroishi/SourceFiles/NHBCSreq20dec2017ah_v21Feb2018/req01102018ah_21feb2018.rdata")

#Loading Key
asvkey = read.csv("/Volumes/f00359t/Dissertation Materials/CordProject/mbl.6W.12M.st.pair.csv",header = FALSE)

#Loading Immune Cell Proportions
immune.original = read.csv("/afs/northstar.dartmouth.edu/udata/margk/collab/AnneHoen/Lab/YukaMoroishi/SourceFiles/NHBCSreq20mar2018ym/NHBCSreq20March2018ym-deided.csv")

#Loading Microbiome Data
setwd("/afs/northstar.dartmouth.edu/udata/margk/collab/AnneHoen/Lab/YukaMoroishi/SourceFiles/JCM_COBR/dada2/sv6W")

#loading New immune cell data 
library(openxlsx)
immune.new = read.xlsx("//afs/northstar.dartmouth.edu/udata/margk/collab/AnneHoen/Lab/YukaMoroishi/SourceFiles/NHBCSreq20mar2018ym/NHBCSreq19SEP2018ym-deided.xlsx")

#Loading Microbiome Data
setwd("/afs/northstar.dartmouth.edu/udata/margk/collab/AnneHoen/Lab/YukaMoroishi/SourceFiles/JCM_COBR/dada2/sv6W")
files = list.files()
files = files[grep(files, pattern = "rds")]
mydata = lapply(files, readRDS)
names(mydata) = files
asvdata.original = mydata[["seqtab.nochim.rds"]]
#Column names of ASV 
asv.colnames = mydata[["seqtab.nochim.colnames.rds"]]
#rm(mydata)
#Genus Level
genusdata.original = mydata[["RDPtaxtab_G_6W.rds"]]

genus.colnames = colnames(genusdata.original)
```

#Data Clean Up
```{r}
immune = immune.original
immune[,17] = as.factor(immune[,17])
asvdata = asvdata.original
genusdata = genusdata.original

immune$mblid = as.character(immune$mblid)
immune$mblid = gsub(" ", "", immune$mblid, fixed = TRUE)
matches = match(immune$mblid, row.names(asvdata))
matches = na.omit(matches)
asvdata = asvdata[matches,]
genusdata = genusdata[matches,]

matches2 = match(row.names(asvdata),immune$mblid)
matches2 = na.omit(matches2)
immune = immune[matches2,]
immune.new2 = immune.new[matches2,]


immune.median = immune[,10:17]
immune.median[,1:7] = apply(immune[,10:16], 2, function(x) ifelse(x>median(x), 1, 0))


immune.median2 = immune.new2[,18:24]
immune.median22 = apply(immune.median2, 2, function(x) ifelse(x>median(x), 1, 0))
```



```{r}
#Phygenetic Tree

source("https://bioconductor.org/biocLite.R")
biocLite("DECIPHER")
biocLite("phangorn")
biocLite("phytools")
biocLite("ips")
biocLite("dada2")
library("DECIPHER")
library("phangorn")
library("phytools")
library("ips")
library("dada2")

####Generating a new phylogenetic tree using fasttree - this works much better ####
#source of assistance: @gitares https://github.com/benjjneb/dada2/issues/88
#http://web.stanford.edu/class/bios221/MicrobiomeWorkflowII.html

asvdata2 = asvdata
colnames(asvdata2) = asv.colnames
seqs.fasttree <- getSequences(asvdata2)
#names(seqs.fasttree) <- asv.colnames
alignment.fasttree <- AlignSeqs(DNAStringSet(seqs.fasttree), anchor = NA, verbose = TRUE)
phagAlign.fasttree <- phyDat(as(alignment.fasttree, "matrix"), type="DNA")
write.phyDat(phagAlign.fasttree, file = "alignment-fasttree.fasta", format = "fasta")

#start terminal and navigate to working directory (which should contain the fasttree program)
#gcc -lm -O2 -Wall -o FastTree FastTree.c
#./FastTree -gtr -nt alignment-fasttree.fasta > fasttree.tre
 
#rooting the tree after using the alignment-fasttree.fasta file to generate a tree from fast tree (tree unrooted)
fasttree.tree <- read.tree("fasttree.tre") #this tree is unrooted
is.rooted(fasttree.tree)
tree.rooted <- midpoint.root(fasttree.tree) #midpoint root from phytools
plot(tree.rooted)
is.rooted(tree.rooted)

library(GUniFrac)
require(ade4)
colnames(asvdata2) = as.character(seq(from = 1, to = ncol(asvdata2), by = 1))
asvrare = Rarefy(asvdata2)$otu.tab.rff
unifracresults = GUniFrac(asvrare, tree.rooted, alpha=c(0, 0.5, 1))$unifracs
dd5 <- unifracresults[, , "d_0.5"]

#ASV Level
#bcell
adonis(as.dist(dd5) ~ immune.median22[,"bcell_revised"])
s.class(cmdscale(dd5, k=2), 
        fac = as.factor(immune.median22[,"bcell_revised"]), 
        col = c("red","blue"),label=invisible())

#cd4t
adonis(as.dist(dd5) ~ immune.median22[,"cd4t_revised"])
s.class(cmdscale(dd5, k=2), fac = as.factor(immune.median22[,"cd4t_revised"]), col = c("red","blue"),label=invisible())

#cd8t
adonis(as.dist(dd5) ~ immune.median22[,"cd8t_revised"])
s.class(cmdscale(dd5, k=2), fac = as.factor(immune.median22[,"cd8t_revised"]), col = c("red","blue"),label=invisible())

#gran
adonis(as.dist(dd5) ~ immune.median22[,"gran_revised"])
s.class(cmdscale(dd5, k=2), fac = as.factor(immune.median22[,"gran_revised"]), col = c("red","blue"),label=invisible())

#mono 
adonis(as.dist(dd5) ~ immune.median22[,"mono_revised"])
s.class(cmdscale(dd5, k=2), fac = as.factor(immune.median22[,"mono_revised"]), col = c("red","blue"),label=invisible())

#nk
adonis(as.dist(dd5) ~ immune.median22[,"nk_revised"])
s.class(cmdscale(dd5, k=2), fac = as.factor(immune.median22[,"nk_revised"]), col = c("red","blue"),label=invisible())

#nrbc
adonis(as.dist(dd5) ~ immune.median22[,"nrbc_revised"])
s.class(cmdscale(dd5, k=2), fac = as.factor(immune.median22[,"nrbc_revised"]), col = c("red","blue"),label=invisible())


```


```{r}
#Using Bray-Curtis
library(vegan)
asvrarebc = Rarefy(asvdata2)$otu.tab.rff
genusrarebc = Rarefy(genusdata)$otu.tab.rff

asvdata2bc = vegdist(asvrarebc, method="bray")
genusdata2bc = vegdist(genusrarebc, method="bray")
asv.bc.clust = hclust(asvdata2bc, method = "complete")

#bcell
adonis(asvdata2bc ~ immune.median22[,"bcell_revised"])
s.class(cmdscale(asvdata2bc, k=2), fac = as.factor(immune.median22[,"bcell_revised"]), col = c("red","blue"),label=invisible())

#cd4t - Marginally. P = 0.106
adonis(asvdata2bc ~ immune.median22[,"cd4t_revised"])
s.class(cmdscale(asvdata2bc, k=2), fac = as.factor(immune.median22[,"cd4t_revised"]), col = c("red","blue"),label=invisible())

#cd8t
adonis(asvdata2bc ~ immune.median22[,"cd8t_revised"])
s.class(cmdscale(asvdata2bc, k=2), fac = as.factor(immune.median22[,"cd8t_revised"]), col = c("red","blue"),label=invisible())

#gran
adonis(asvdata2bc ~ immune.median22[,"gran_revised"])
s.class(cmdscale(asvdata2bc, k=2), fac = as.factor(immune.median22[,"gran_revised"]), col = c("red","blue"),label=invisible())

#mono
adonis(asvdata2bc ~ immune.median22[,"mono_revised"])
s.class(cmdscale(asvdata2bc, k=2), fac = as.factor(immune.median22[,"mono_revised"]), col = c("red","blue"),label=invisible())

#nk - P = 0.149
adonis(asvdata2bc ~ immune.median22[,"nk_revised"])
s.class(cmdscale(asvdata2bc, k=2), fac = as.factor(immune.median22[,"nk_revised"]), col = c("red","blue"),label=invisible())

#nrbc
adonis(asvdata2bc ~ immune.median22[,"nrbc_revised"])
s.class(cmdscale(asvdata2bc, k=2), fac = as.factor(immune.median22[,"nrbc_revised"]), col = c("red","blue"),label=invisible())


```

#Genus level
```{r}
genusrarebc = Rarefy(genusdata)$otu.tab.rff
genusdata2bc = vegdist(genusrarebc, method="bray")


#bcell
adonis(genusdata2bc ~ immune.median22[,"bcell_revised"])
s.class(cmdscale(genusdata2bc, k=2), fac = as.factor(immune.median22[,"bcell_revised"]), col = c("red","blue"),label=invisible())

#cd4t - Not significant at genus level
adonis(genusdata2bc ~ immune.median22[,"cd4t_revised"])
s.class(cmdscale(genusdata2bc, k=2), fac = as.factor(immune.median22[,"cd4t_revised"]), col = c("red","blue"),label=invisible())

#cd8t
adonis(genusdata2bc ~ immune.median22[,"cd8t_revised"])
s.class(cmdscale(genusdata2bc, k=2), fac = as.factor(immune.median22[,"cd8t_revised"]), col = c("red","blue"),label=invisible())

#gran
adonis(genusdata2bc ~ immune.median22[,"gran_revised"])
s.class(cmdscale(genusdata2bc, k=2), fac = as.factor(immune.median22[,"gran_revised"]), col = c("red","blue"),label=invisible())

#mono
adonis(genusdata2bc ~ immune.median22[,"mono_revised"])
s.class(cmdscale(genusdata2bc, k=2), fac = as.factor(immune.median22[,"mono_revised"]), col = c("red","blue"),label=invisible())

#nk - P = 0.149
adonis(genusdata2bc ~ immune.median22[,"nk_revised"])
s.class(cmdscale(genusdata2bc, k=2), fac = as.factor(immune.median22[,"nk_revised"]), col = c("red","blue"),label=invisible())

#nrbc
adonis(genusdata2bc ~ immune.median22[,"nrbc_revised"])
s.class(cmdscale(genusdata2bc, k=2), fac = as.factor(immune.median22[,"nrbc_revised"]), col = c("red","blue"),label=invisible())


```
#Alpha diversity
```{r}
H <- diversity(asvdata)
simp <- diversity(asvdata, "simpson")
invsimp <- diversity(asvdata, "inv")
## Unbiased Simpson (Hurlbert 1971, eq. 5) with rarefy:
unbias.simp <- rarefy(asvdata, 2) - 1
## Fisher alpha
alpha <- fisher.alpha(asvdata)
## Plot all
pairs(cbind(H, simp, invsimp, unbias.simp, alpha), pch="+", col="blue")

#Shannon Index
t.test(diversity(asvdata)~immune.median22[,"bcell_revised"])
t.test(diversity(asvdata)~immune.median22[,"cd4t_revised"])
t.test(diversity(asvdata)~immune.median22[,"cd8t_revised"])
t.test(diversity(asvdata)~immune.median22[,"mono_revised"])
t.test(diversity(asvdata)~immune.median22[,"gran_revised"])
t.test(diversity(asvdata)~immune.median22[,"nk_revised"])
t.test(diversity(asvdata)~immune.median22[,"nrbc_revised"])

par(mfrow=c(2,4))
boxplot(diversity(asvdata)~immune.median22[,"bcell_revised"], 
        ylab = "H", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "Bcell")


boxplot(diversity(asvdata)~immune.median22[,"cd4t_revised"], 
        ylab = "H", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "CD4T")

boxplot(diversity(asvdata)~immune.median22[,"cd8t_revised"], 
        ylab = "H", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "CD8T")

boxplot(diversity(asvdata)~immune.median22[,"mono_revised"], 
        ylab = "H", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "Monocyte")

boxplot(diversity(asvdata)~immune.median22[,"gran_revised"], 
        ylab = "H", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "Granulocyte")

boxplot(diversity(asvdata)~immune.median22[,"nk_revised"], 
        ylab = "H", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "Natural Killer")

boxplot(diversity(asvdata)~immune.median22[,"nrbc_revised"], 
        ylab = "H", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "nRBC")


#Simpsom Index
t.test(diversity(asvdata, "simpson")~immune.median22[,"bcell_revised"])
t.test(diversity(asvdata, "simpson")~immune.median22[,"cd4t_revised"])
t.test(diversity(asvdata, "simpson")~immune.median22[,"cd8t_revised"])
t.test(diversity(asvdata, "simpson")~immune.median22[,"mono_revised"])
t.test(diversity(asvdata, "simpson")~immune.median22[,"gran_revised"])
t.test(diversity(asvdata, "simpson")~immune.median22[,"nk_revised"])
t.test(diversity(asvdata, "simpson")~immune.median22[,"nrbc_revised"])

par(mfrow=c(2,4))
boxplot(diversity(asvdata, "simpson")~immune.median22[,"bcell_revised"], 
        ylab = "D", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "Bcell")


boxplot(diversity(asvdata, "simpson")~immune.median22[,"cd4t_revised"], 
        ylab = "D", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "CD4T")

boxplot(diversity(asvdata, "simpson")~immune.median22[,"cd8t_revised"], 
        ylab = "D", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "CD8T")

boxplot(diversity(asvdata, "simpson")~immune.median22[,"mono_revised"], 
        ylab = "D", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "Monocyte")

boxplot(diversity(asvdata, "simpson")~immune.median22[,"gran_revised"], 
        ylab = "D", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "Granulocyte")

boxplot(diversity(asvdata, "simpson")~immune.median22[,"nk_revised"], 
        ylab = "D", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "Natural Killer")

boxplot(diversity(asvdata, "simpson")~immune.median22[,"nrbc_revised"], 
        ylab = "D", xlab = "Immune Cell Proportion",
        names=c(">=Median", "<Median"),
        main = "nRBC")
```


#12M
```{r}
#Loading Microbiome Data
setwd("/afs/northstar.dartmouth.edu/udata/margk/collab/AnneHoen/Lab/YukaMoroishi/SourceFiles/JCM_COBR/dada2/sv12M")

#loading New immune cell data 

files = list.files()
files = files[grep(files, pattern = "rds")]
mydata = lapply(files, readRDS)
names(mydata) = files
asvdata.original12 = mydata[["seqtab.nochim.rds"]]
#Column names of ASV 
asv.colnames12 = mydata[["seqtab.nochim.colnames.rds"]]
#rm(mydata)
#Genus Level
genusdata.original12 = mydata[["RDPtaxtab_G_12M.rds"]]

genus.colnames12 = colnames(genusdata.original12)

asvdata12M = asvdata.original12
genusdata12M = genusdata.original12

#immune$mblid = as.character(immune$mblid)
#immune$mblid = gsub(" ", "", immune$mblid, fixed = TRUE)
immune.new$mblid = gsub("    ", "", immune.new$mblid, fixed = TRUE)

#mblid2match = match(immune.new$mblid, asvkey$V2)
#mblid2 = asvkey$V3[mblid2match]


matches12M = match(immune.new$mblid, row.names(asvdata12M))
matches12M = na.omit(matches12M)
asvdata12M = asvdata12M[matches12M,]
genusdata12M = genusdata12M[matches12M,]

matches212M = match(row.names(asvdata12M),immune.new$mblid)
matches212M = na.omit(matches212M)
immune.new212M = immune.new[matches212M,]


asvkey$V2[na.omit(match(row.names(asvdata12M), asvkey$V3))]
length(na.omit(match(row.names(asvdata), asvkey$V2[na.omit(match(row.names(asvdata12M), asvkey$V3))])))
```

#Correlation
```{r}
library(ggcorrplot)

cor.im6 = cor(cbind(immune.new2[,18:24], genusdata.nozeros))
row.names(cor.im6) = c("bcell", "cd4t", "cd8t", "gran", "mono", "nk", 
                       "nrbc",
                       as.character(seq(from = 1, 
                                        to = ncol(genusdata.nozeros),
                                        by = 1)))

colnames(cor.im6) = row.names(cor.im6)

ggcorrplot(cor.im6[8:nrow(cor.im6),1:7])
```

#MZILN
```{r}
library(devtools)
install_github("quangnguyen1995/MZILN")
library(MZILN)
library(utils)

MZILN.main(df.main, df.covar, covariates = colnames(df.covar),
  n.lam = 200, lam.min.ratio = 1e-04, reg.method = "mcp",
  n.folds = 5)


df.cd4t = as.data.frame(immune.new2[,"cd4t_revised"])
colnames(df.cd4t) = c("cd4t")

genusdata.nozeros = genusdata[,-which(colSums(genusdata) == 0)]
genusdata.nozeros.prop = prop.table(as.matrix(genusdata.nozeros), margin=1)


MZILNresults = MZILN.main(df.main = genusdata.nozeros.prop, 
                          df.covar = df.cd4t, 
           covariates = colnames(df.cd4t),
           n.lam = 200, lam.min.ratio = 1e-04, reg.method = "mcp",
           n.folds = 5)


Find.Taxa2(df.cd4t, genusdata.nozeros.prop)


get.coef.mziln = function(data, taxavec, df.covariates, indices){
  matrix.data.full = data[indices,2:ncol(data)]
  log.data.full = data[indices,1]
  #taxa.vec.ind = taxavec[indices,]
  taxa.vec.ind = taxavec
  test2 = cv.ncvreg(X = matrix.data.full, y = log.data.full, standardize = FALSE, 
                    penalty="MCP", returnX=FALSE, nfolds = 5)
  results = as.vector(coef(test2, s = "lambda.min"))
  result <- split(results, ceiling(seq_along(results)/(ncol(df.covariates) + 1)))
  result <- as.data.frame(do.call("rbind",result))
  result = result[-nrow(result),]
  rownames(result) <- colnames(taxa.vec.ind )[-length(colnames(taxa.vec.ind ))]
  colnames(result) <- c("Intercept",colnames(df.covariates))
  coefvec = as.vector(result[,2])
  return(coefvec)
}

library(picasso)
get.coef.mziln.biglasso = function(data, taxavec, df.covariates, indices){
  matrix.data.full = data[indices,2:ncol(data)]
  log.data.full = data[indices,1]
  #taxa.vec.ind = taxavec[indices,]
  taxa.vec.ind = taxavec
  #test2 = cv.ncvreg(X = matrix.data.full, y = log.data.full, standardize = FALSE, penalty="MCP", returnX=FALSE, nfolds = 5)
  test2 = cv.biglasso(X = matrix.data.full, y = log.data.full)
  picasso.data <- picasso(X = matrix.data.full,Y = log.data.full,
                          lambda = NULL, nlambda = n.lam, 
                          family = 'gaussian',
                          method = "mcp",type.gaussian = 'naive',
                          standardize = FALSE, 
                          verbose = FALSE,
                          lambda.min.ratio = lam.min.ratio)

  lambda.values <- picasso.data$lambda #extracting lambda values
  shuffle <- sample(nrow(matrix.data.full)) #creating a shuffle pattern
  log.data.full <- log.data.full[shuffle] #shuffle the log transformed data set.
  matrix.data.full <- matrix.data.full[shuffle,] #shuffle matrix x similar to the one before
  n.folds = 5
  folds <- cut(seq(1:nrow(matrix.data.full)),breaks=n.folds,labels=FALSE)
  lambda.error <- matrix(nrow = n.lam, ncol = n.folds)
  for(i in (1:n.folds)){
    #Segement your data by fold using the which() function
    testIndexes <- which(folds==i,arr.ind=TRUE)
    test.log <- log.data.full[testIndexes]
    test.covar <- matrix.data.full[testIndexes,]
    train.log <- log.data.full[-testIndexes]
    train.covar <- matrix.data.full[-testIndexes,]
    #set it to  lambda.values
    picasso.train <- picasso::picasso(X = train.covar,Y = train.log,
                             lambda = lambda.values, family = 'gaussian',
                             method = "mcp", type.gaussian = 'naive',
                             standardize = FALSE, verbose = FALSE)
    f <- file()
    sink(file = f)
    predict <- picasso::predict.gaussian(picasso.train, test.covar,
                                lambda.idx = c(1:n.lam), 
                                y.pred.idx = c(1:n.lam))
    sink()
    close(f)
    for (l in (1:n.lam)){
      lambda.error[l,i] <- (sum((predict[,l] - test.log)^2)) / nrow(as.matrix(test.log) )
    }
  }
  
  lambda.error <- as.data.frame(lambda.error)
  lambda.error$mean <- rowSums(lambda.error)/n.folds
  lambda.error$lambda <- lambda.values
  #### end cross validation ####
  v <- which(lambda.error$mean == min(lambda.error$mean)) #lambda value with the lowest
  if(v == row(lambda.error$mean)){}
  beta <- as.vector(picasso.data$beta[,v])
  #generating the results
  list <- list() #empty list
  #result
result <- split(beta, ceiling(seq_along(beta)/(ncol(df.covariates) + 1)))
  
  
  results = as.vector(coef(test2, s = "lambda.min"))
  result <- split(results, ceiling(seq_along(results)/(ncol(df.covariates) + 1)))
  result <- as.data.frame(do.call("rbind",result))
  result = result[-nrow(result),]
  rownames(result) <- colnames(taxa.vec.ind )[-length(colnames(taxa.vec.ind ))]
  colnames(result) <- c("Intercept",colnames(df.covariates))
  coefvec = as.vector(result[,2])
  return(coefvec)
}


testtr.new = Transform.Data(df.cd4t, genusdata.nozeros.prop)
library(boot)
set.seed(123)
system.time(
results2.1 <- boot(data=testtr.new, taxavec = genusdata.nozeros.prop,
                df.covariates = df.cd4t,
                statistic=get.coef.mziln, 
  	R=100)
)


tot.df2 = c()
for (i in 1:194){
  num = length(which(results2.1$t[,i]!=0))
  if (num > 70)
  {
    tot.df2 = c(tot.df2, i)
 }}
length(tot.df2)

colnames(genusdata.nozeros.prop[,tot.df])
colnames(genusdata.nozeros.prop[,tot.df2])

```

#Select reference taxa
```{r}
library(dplyr)
genusdata.nozeros.prop.01 = ifelse(genusdata.nozeros.prop>0,1,0)
genusdata.nozeros.prop.colsum = colSums(genusdata.nozeros.prop.01)
hist(genusdata.nozeros.prop.colsum)

randsamp10asvcd4t = sample(which(genusdata.nozeros.prop.colsum>= 10 & genusdata.nozeros.prop.colsum <=35), size = 10)
genusdata.nozeros.prop.2 = genusdata.nozeros.prop
genusdata.nozeros.prop.2 = as.data.frame(genusdata.nozeros.prop.2)

#genusdata.nozeros.prop.2.1 = genusdata.nozeros.prop.2 %>% select(-Pseudomonas,everything())

#Pseudomonas
ref1 = as.data.frame(genusdata.nozeros.prop.2$Pseudomonas)
colnames(ref1) = "Pseudomonas"
genusdata.nozeros.prop.2.1 =  genusdata.nozeros.prop.2[,-which(colnames(genusdata.nozeros.prop.2)=="Pseudomonas")]

genusdata.nozeros.prop.2.1 = cbind(genusdata.nozeros.prop.2.1, ref1)

testtr.new2 = Transform.Data(df.cd4t, genusdata.nozeros.prop.2.1)
system.time(
results2 <- boot(data=testtr.new2, taxavec = genusdata.nozeros.prop.2.1,
                df.covariates = df.cd4t,
                statistic=get.coef.mziln, 
  	R=100)
)

tot.df = c()
for (i in 1:194){
  num = length(which(results2$t[,i]!=0))
  if (num > 70)
  {
    tot.df = c(tot.df, i)
 }}
length(tot.df)

colnames(genusdata.nozeros.prop.2.1[,tot.df])
colnames(genusdata.nozeros.prop[,tot.df])



#Atopobium
ref2 = as.data.frame(genusdata.nozeros.prop.2$Atopobium)
colnames(ref2) = "Atopobium"
genusdata.nozeros.prop.2.2 =  genusdata.nozeros.prop.2[,-which(colnames(genusdata.nozeros.prop.2)=="Atopobium")]

genusdata.nozeros.prop.2.2 = cbind(genusdata.nozeros.prop.2.2, ref2)

testtr.new3 = Transform.Data(df.cd4t, genusdata.nozeros.prop.2.2)

set.seed(1234)
system.time(
results2.2 <- boot(data=testtr.new3, taxavec = genusdata.nozeros.prop.2.2,
                df.covariates = df.cd4t,
                statistic=get.coef.mziln, 
  	R=100)
)


tot.df3 = c()
for (i in 1:194){
  num = length(which(results2.2$t[,i]!=0))
  if (num > 70)
  {
    tot.df3 = c(tot.df3, i)
 }}
length(tot.df3)

colnames(genusdata.nozeros.prop.2.2[,tot.df3])




#Acinetobacter
ref3 = as.data.frame(genusdata.nozeros.prop.2$Acinetobacter)
colnames(ref3) = "Acinetobacter"
genusdata.nozeros.prop.2.3 =  genusdata.nozeros.prop.2[,-which(colnames(genusdata.nozeros.prop.2)=="Acinetobacter")]

genusdata.nozeros.prop.2.3 = cbind(genusdata.nozeros.prop.2.3, ref3)

testtr.new4 = Transform.Data(df.cd4t, genusdata.nozeros.prop.2.3)

set.seed(12345)
system.time(
results2.3 <- boot(data=testtr.new4, taxavec = genusdata.nozeros.prop.2.3,
                df.covariates = df.cd4t,
                statistic=get.coef.mziln, 
  	R=100)
)


tot.df4 = c()
for (i in 1:194){
  num = length(which(results2.3$t[,i]!=0))
  if (num > 70)
  {
    tot.df4 = c(tot.df4, i)
 }}
length(tot.df4)

colnames(genusdata.nozeros.prop.2.3[,tot.df4])



#Rothia
ref5 = as.data.frame(genusdata.nozeros.prop.2$Rothia)
colnames(ref5) = "Rothia"
genusdata.nozeros.prop.2.5 =  genusdata.nozeros.prop.2[,-which(colnames(genusdata.nozeros.prop.2)=="Rothia")]

genusdata.nozeros.prop.2.5 = cbind(genusdata.nozeros.prop.2.5, ref5)

testtr.new5 = Transform.Data(df.cd4t, genusdata.nozeros.prop.2.5)

set.seed(123456)
system.time(
results2.5 <- boot(data=testtr.new5, taxavec = genusdata.nozeros.prop.2.5,
                df.covariates = df.cd4t,
                statistic=get.coef.mziln, 
  	R=100)
)


tot.df5 = c(NULL)
for (i in 1:194){
  num = length(which(results2.5$t[,i]!=0))
  if (num > 70)
  {
    tot.df5 = c(tot.df5, i)
 }}
length(tot.df5)

colnames(genusdata.nozeros.prop.2.5[,tot.df5])



#All together
a1 = colnames(genusdata.nozeros.prop.2.1[,tot.df])
a2 = colnames(genusdata.nozeros.prop[,tot.df2])
a3 = colnames(genusdata.nozeros.prop.2.2[,tot.df3])
a4 = colnames(genusdata.nozeros.prop.2.3[,tot.df4])
a5 = colnames(genusdata.nozeros.prop.2.5[,tot.df5])





cd4t.genera.selection = Reduce(intersect, list(a1,a2, a3, a4, a5))

save(cd4t.genera.selection, a1, a2, a3, a4, a5,results2.5, results2.3, results2.2,results2,results2.1, file = "cd4tdataresults.RData")
```

CD4T:

 [1] "Bacteroides"               "Bifidobacterium"          
 [3] "Citrobacter"               "Clostridium_sensu_stricto"
 [5] "Enterobacter"              "Enterococcus"             
 [7] "Escherichia/Shigella"      "F__Enterobacteriaceae"    
 [9] "Flavonifractor"            "Klebsiella"               
[11] "Parabacteroides"           "Streptococcus"  


cd4t --> a specific subset related to diversity in microbiome. TH..?

#ASV level cd4t
```{r}
testtr.asv = Transform.Data(df.cd4t, asvdata2)
set.seed(123)
system.time(
resultsasv <- boot(data=testtr.asv, taxavec = asvdata2,
                df.covariates = df.cd4t,
                statistic=get.coef.mziln, 
  	R=3)
)

tot.dfasv = c()
for (i in 1:(ncol(asvdata2)-1)){
  num = length(which(resultsasv$t[,i]!=0))
  if (num > 70)
  {
    tot.dfasv  = c(tot.dfasv , i)
 }}
length(tot.dfasv)

colnames(genusdata.nozeros.prop.2.3[,tot.dfasv ])


```


```{r}
#Creating files
df.cd4t = as.data.frame(immune.new2[,"cd4t_revised"])
colnames(df.cd4t) = c("cd4t")
df.cd8t = as.data.frame(immune.new2[,"cd8t_revised"])
colnames(df.cd8t) = c("cd8t")
df.bcell = as.data.frame(immune.new2[,"bcell_revised"])
colnames(df.bcell) = c("bcell")
df.gran = as.data.frame(immune.new2[,"gran_revised"])
colnames(df.gran) = c("gran")
df.mono = as.data.frame(immune.new2[,"mono_revised"])
colnames(df.mono) = c("mono")
df.nk = as.data.frame(immune.new2[,"nk_revised"])
colnames(df.nk) = c("nk")
df.nrbc = as.data.frame(immune.new2[,"nrbc_revised"])
colnames(df.nrbc) = c("nrbc")



save(get.coef.mziln,Transform.Data, genusdata.nozeros.prop.2, df.cd4t,df.cd8t, df.bcell, log.trans.generation, matrix.a.generation,matrix.x.generation, defining.row.1, defining.row.2 , 
     df.gran, df.mono, df.nk, df.nrbc, file = "inputdata.RData")
```


```{r}
#Getting all reference taxa

#load("/Volumes/rc/genusjobs/tdfcd4t1.RData")
#tdf

#Cd4t
selected.cd4tall = matrix(0, nrow = 195, ncol = 195)
row.names(selected.cd4tall) = colnames(genusdata.nozeros)
colnames(selected.cd4tall) = colnames(genusdata.nozeros)
genuscd4tlist = list()

for(i in 1:195){
  print(i)
  load(paste("/Volumes/rc/genusjobs/cd4tresults/tdfcd4t", as.character(i), ".RData", sep = ""))
  genuscd4tlist[[i]] = tdf
  match.select = match(row.names(tdf), colnames(genusdata.nozeros))
  selected.cd4tall[i,match.select] = selected.cd4tall[i,match.select] + 1
}

#Each row is sample

which(colSums(selected.cd4tall) > (195*.7))
which(colSums(selected.cd4tall) > (195*.8))
which(colSums(selected.cd4tall) > (195*.9))


#Cd8t
selected.cd8tall = matrix(0, nrow = 195, ncol = 195)
row.names(selected.cd8tall) = colnames(genusdata.nozeros)
colnames(selected.cd8tall) = colnames(genusdata.nozeros)
genuscd8tlist = list()

for(i in 1:195){
  print(i)
  load(paste("/Volumes/rc/genusjobs/cd8tresults/tdfcd8t", as.character(i), ".RData", sep = ""))
  genuscd8tlist[[i]] = tdf
  match.select = match(row.names(tdf), colnames(genusdata.nozeros))
  selected.cd8tall[i,match.select] = selected.cd8tall[i,match.select] + 1
}


#Each row is sample

which(colSums(selected.cd8tall) > (195*.7))
which(colSums(selected.cd8tall) > (195*.8))
which(colSums(selected.cd8tall) > (195*.9))



#bcell
selected.bcellall = matrix(0, nrow = 195, ncol = 195)
row.names(selected.bcellall) = colnames(genusdata.nozeros)
colnames(selected.bcellall) = colnames(genusdata.nozeros)
genusbcelllist = list()

for(i in 1:195){
  print(i)
  load(paste("/Volumes/rc/genusjobs/bcellresults/tdfbcell", as.character(i), ".RData", sep = ""))
  genusbcelllist[[i]] = tdf
  match.select = match(row.names(tdf), colnames(genusdata.nozeros))
  selected.bcellall[i,match.select] = selected.bcellall[i,match.select] + 1
}

#Each row is sample

which(colSums(selected.bcellall) > (195*.7))
which(colSums(selected.bcellall) > (195*.8))
which(colSums(selected.bcellall) > (195*.9))


#gran
selected.granall = matrix(0, nrow = 195, ncol = 195)
row.names(selected.granall) = colnames(genusdata.nozeros)
colnames(selected.granall) = colnames(genusdata.nozeros)
genusgranlist = list()

for(i in 1:195){
  print(i)
  load(paste("/Volumes/rc/genusjobs/granresults/tdfgran", as.character(i), ".RData", sep = ""))
  genusgranlist[[i]] = tdf
  match.select = match(row.names(tdf), colnames(genusdata.nozeros))
  selected.granall[i,match.select] = selected.granall[i,match.select] + 1
}

#Each row is sample

which(colSums(selected.granall) > (195*.7))
which(colSums(selected.granall) > (195*.8))
which(colSums(selected.granall) > (195*.9))


#mono
selected.monoall = matrix(0, nrow = 195, ncol = 195)
row.names(selected.monoall) = colnames(genusdata.nozeros)
colnames(selected.monoall) = colnames(genusdata.nozeros)
genusmonolist = list()

for(i in 1:195){
  print(i)
  load(paste("/Volumes/rc/genusjobs/monoresults/tdfmono", as.character(i), ".RData", sep = ""))
  genusmonolist[[i]] = tdf
  match.select = match(row.names(tdf), colnames(genusdata.nozeros))
  selected.monoall[i,match.select] = selected.monoall[i,match.select] + 1
}

#Each row is sample

which(colSums(selected.monoall) > (195*.7))
which(colSums(selected.monoall) > (195*.8))
which(colSums(selected.monoall) > (195*.9))


#nk
selected.nkall = matrix(0, nrow = 195, ncol = 195)
row.names(selected.nkall) = colnames(genusdata.nozeros)
colnames(selected.nkall) = colnames(genusdata.nozeros)
genusnklist = list()

for(i in 1:195){
  print(i)
  load(paste("/Volumes/rc/genusjobs/nkresults/tdfnk", as.character(i), ".RData", sep = ""))
  genusnklist[[i]] = tdf
  match.select = match(row.names(tdf), colnames(genusdata.nozeros))
  selected.nkall[i,match.select] = selected.nkall[i,match.select] + 1
}

#Each row is sample

which(colSums(selected.nkall) > (195*.7))
which(colSums(selected.nkall) > (195*.8))
which(colSums(selected.nkall) > (195*.9))


#nrbc
selected.nrbcall = matrix(0, nrow = 195, ncol = 195)
row.names(selected.nrbcall) = colnames(genusdata.nozeros)
colnames(selected.nrbcall) = colnames(genusdata.nozeros)
genusnrbclist = list()

for(i in 1:195){
  print(i)
  load(paste("/Volumes/rc/genusjobs/nrbcresults/tdfnrbc", as.character(i), ".RData", sep = ""))
  genusnrbclist[[i]] = tdf
  match.select = match(row.names(tdf), colnames(genusdata.nozeros))
  selected.nrbcall[i,match.select] = selected.nrbcall[i,match.select] + 1
}

which(colSums(selected.nrbcall) > (195*.7))
which(colSums(selected.nrbcall) > (195*.8))
which(colSums(selected.nrbcall) > (195*.9))


summary(lm(newgenusdata$Bacteroides ~ immune.new2$nrbc))
summary(lm(newgenusdata$Bifidobacterium ~ immune.new2$nrbc))
summary(lm(newgenusdata$Clostridium_sensu_stricto ~ immune.new2$nrbc))
summary(lm(newgenusdata$Enterococcus ~ immune.new2$nrbc))

```




Visualizing
```{r}
library(UpSetR)


listInput2 = list(bcell = colnames(genusdata.nozeros)[which(colSums(selected.bcellall) > (195*.9))], 
                cd4t = colnames(genusdata.nozeros)[which(colSums(selected.cd4tall) > (195*.9))],
                cd8t = colnames(genusdata.nozeros)[which(colSums(selected.cd8tall) > (195*.9))],
                nk = colnames(genusdata.nozeros)[which(colSums(selected.nkall) > (195*.9))],
                mono = colnames(genusdata.nozeros)[which(colSums(selected.monoall) > (195*.9))], 
                gran = colnames(genusdata.nozeros)[which(colSums(selected.granall) > (195*.9))],
                nrbc = colnames(genusdata.nozeros)[which(colSums(selected.nrbcall) > (195*.9))])

upset(fromList(listInput2), order.by = "freq", nsets = 7, empty.intersections = "on")


bcellseltaxgen = colnames(genusdata.nozeros)[which(colSums(selected.bcellall) > (195*.9))]
cd4tseltaxgen = colnames(genusdata.nozeros)[which(colSums(selected.cd4tall) > (195*.9))]
cd8tseltaxgen = colnames(genusdata.nozeros)[which(colSums(selected.cd8tall) > (195*.9))]
granseltaxgen = colnames(genusdata.nozeros)[which(colSums(selected.granall) > (195*.9))]
monoseltaxgen = colnames(genusdata.nozeros)[which(colSums(selected.monoall) > (195*.9))]
nkseltaxgen = colnames(genusdata.nozeros)[which(colSums(selected.nkall) > (195*.9))]
nrbcseltaxgen = colnames(genusdata.nozeros)[which(colSums(selected.nrbcall) > (195*.9))]

unique(unlist(listInput2))
length(unique(unlist(listInput2)))


immune.crude.results = matrix(0, nrow = 7, ncol = 11)
colnames(immune.crude.results) = unique(unlist(listInput2))
row.names(immune.crude.results) = c("bcell", "cd4t", "cd8t", "gran", "mono", "nk", "nrbc")


immune.crude.results[1,] = as.numeric(unique(unlist(listInput2)) %in% bcellseltaxgen)
immune.crude.results[2,] = as.numeric(unique(unlist(listInput2)) %in% cd4tseltaxgen)
immune.crude.results[3,] = as.numeric(unique(unlist(listInput2)) %in% cd8tseltaxgen)
immune.crude.results[4,] = as.numeric(unique(unlist(listInput2)) %in% granseltaxgen)
immune.crude.results[5,] = as.numeric(unique(unlist(listInput2)) %in% monoseltaxgen)
immune.crude.results[6,] = as.numeric(unique(unlist(listInput2)) %in% nkseltaxgen)
immune.crude.results[7,] = as.numeric(unique(unlist(listInput2)) %in% nrbcseltaxgen)

immune.crude.results
upset(fromList(listInput2), order.by = "freq", nsets = 7)

```



#Adjusted Analysis
```{r}
#Imputing missing data for ever cig
library(mice)
dataimp = as.data.frame(immune.new2[,c("cd4t_revised", "cd8t_revised", "bcell_revised", "gran_revised", "mono_revised", "nk_revised", "nrbc_revised", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")])
tempData <- mice(dataimp,m=5,maxit=50,meth='pmm',seed=500)
completedData <- complete(tempData,1)





#Creating files for adjusted analyses
df.cd4tadj = as.data.frame(completedData[,c("cd4t_revised", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")])
colnames(df.cd4tadj) = c("cd4t", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")
df.cd8tadj = as.data.frame(completedData[,c("cd8t_revised", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")])
colnames(df.cd8tadj) = c("cd8t", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")
df.bcelladj = as.data.frame(completedData[,c("bcell_revised", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")])
colnames(df.bcelladj) = c("bcell", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")
df.granadj = as.data.frame(completedData[,c("gran_revised", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")])
colnames(df.granadj) = c("gran", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")
df.monoadj = as.data.frame(completedData[,c("mono_revised", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")])
colnames(df.monoadj) = c("mono", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")
df.nkadj = as.data.frame(completedData[,c("nk_revised", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")])
colnames(df.nkadj) = c("nk", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")
df.nrbcadj = as.data.frame(completedData[,c("nrbc_revised", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")])
colnames(df.nrbcadj) = c("nrbc", "bbymale", "enrollment_age",  "imrbthwghtg_all", "gestage_all", "evercigpreg")

save(get.coef.mziln,Transform.Data, genusdata.nozeros.prop.2, df.cd4tadj,df.cd8tadj, df.bcelladj, log.trans.generation, matrix.a.generation,matrix.x.generation, defining.row.1, defining.row.2 , 
     df.granadj, df.monoadj, df.nkadj, df.nrbcadj, file = "/Volumes/rc/genusjobs/inputdataadj.RData")

save(df.bcelladj, genusdata.nozeros.prop.2, genusdata.nozeros, df.cd4tadj, df.cd8tadj, df.granadj, df.monoadj, df.nkadj, df.nrbcadj,df.cd8tcd4tadj, file = "inputdataMZILN.Rdata")

#bcell
selected.bcellalladj = matrix(0, nrow = 195, ncol = 195)
row.names(selected.bcellalladj) = colnames(genusdata.nozeros)
colnames(selected.bcellalladj) = colnames(genusdata.nozeros)
genusbcelllistadj = list()

for(i in 1:195){
  print(i)
  load(paste("/Volumes/rc/home/t/f00359t/genusjobs/bcellresultsadj/tdfbcelladj", as.character(i), ".RData", sep = ""))
  genusbcelllistadj[[i]] = tdf
  match.select = match(row.names(tdf), colnames(genusdata.nozeros))
  selected.bcellalladj[i,match.select] = selected.bcellalladj[i,match.select] + 1
}

#Each row is sample

which(colSums(selected.bcellalladj) > (195*.7))
which(colSums(selected.bcellalladj) > (195*.8))
which(colSums(selected.bcellalladj) > (195*.9))



#cd4t
selected.cd4talladj = matrix(0, nrow = 195, ncol = 195)
row.names(selected.cd4talladj) = colnames(genusdata.nozeros)
colnames(selected.cd4talladj) = colnames(genusdata.nozeros)
genuscd4tlistadj = list()

for(i in 1:195){
  print(i)
  load(paste("/Volumes/rc/home/t/f00359t/genusjobs/cd4tresultsadj/tdfcd4tadj", as.character(i), ".RData", sep = ""))
  genuscd4tlistadj[[i]] = tdf
  match.select = match(row.names(tdf), colnames(genusdata.nozeros))
  selected.cd4talladj[i,match.select] = selected.cd4talladj[i,match.select] + 1
}

#Each row is sample

which(colSums(selected.cd4talladj) > (195*.7))
which(colSums(selected.cd4talladj) > (195*.8))
which(colSums(selected.cd4talladj) > (195*.9))


#cd8t
selected.cd8talladj = matrix(0, nrow = 195, ncol = 195)
row.names(selected.cd8talladj) = colnames(genusdata.nozeros)
colnames(selected.cd8talladj) = colnames(genusdata.nozeros)
genuscd8tlistadj = list()

for(i in 1:195){
  print(i)
  load(paste("/Volumes/rc/genusjobs/cd8tresultsadj/tdfcd8tadj", as.character(i), ".RData", sep = ""))
  genuscd8tlistadj[[i]] = tdf
  match.select = match(row.names(tdf), colnames(genusdata.nozeros))
  selected.cd8talladj[i,match.select] = selected.cd8talladj[i,match.select] + 1
}

#Each row is sample

which(colSums(selected.cd8talladj) > (195*.7))
which(colSums(selected.cd8talladj) > (195*.8))
which(colSums(selected.cd8talladj) > (195*.9))


#gran
selected.granalladj = matrix(0, nrow = 195, ncol = 195)
row.names(selected.granalladj) = colnames(genusdata.nozeros)
colnames(selected.granalladj) = colnames(genusdata.nozeros)
genusgranlistadj = list()

for(i in 1:195){
  print(i)
  load(paste("/Volumes/rc/genusjobs/granresultsadj/tdfgranadj", as.character(i), ".RData", sep = ""))
  genusgranlistadj[[i]] = tdf
  match.select = match(row.names(tdf), colnames(genusdata.nozeros))
  selected.granalladj[i,match.select] = selected.granalladj[i,match.select] + 1
}

#Each row is sample

which(colSums(selected.granalladj) > (195*.7))
which(colSums(selected.granalladj) > (195*.8))
which(colSums(selected.granalladj) > (195*.9))

#mono
selected.monoalladj = matrix(0, nrow = 195, ncol = 195)
row.names(selected.monoalladj) = colnames(genusdata.nozeros)
colnames(selected.monoalladj) = colnames(genusdata.nozeros)
genusmonolistadj = list()

for(i in 1:195){
  print(i)
  load(paste("/Volumes/rc/genusjobs/monoresultsadj/tdfmonoadj", as.character(i), ".RData", sep = ""))
  genusmonolistadj[[i]] = tdf
  match.select = match(row.names(tdf), colnames(genusdata.nozeros))
  selected.monoalladj[i,match.select] = selected.monoalladj[i,match.select] + 1
}

#Each row is sample

which(colSums(selected.monoalladj) > (195*.7))
which(colSums(selected.monoalladj) > (195*.8))
which(colSums(selected.monoalladj) > (195*.9))

#nk
selected.nkalladj = matrix(0, nrow = 195, ncol = 195)
row.names(selected.nkalladj) = colnames(genusdata.nozeros)
colnames(selected.nkalladj) = colnames(genusdata.nozeros)
genusnklistadj = list()

for(i in 1:195){
  print(i)
  load(paste("/Volumes/rc/genusjobs/nkresultsadj/tdfnkadj", as.character(i), ".RData", sep = ""))
  genusnklistadj[[i]] = tdf
  match.select = match(row.names(tdf), colnames(genusdata.nozeros))
  selected.nkalladj[i,match.select] = selected.nkalladj[i,match.select] + 1
}

#Each row is sample

which(colSums(selected.nkalladj) > (195*.7))
which(colSums(selected.nkalladj) > (195*.8))
which(colSums(selected.nkalladj) > (195*.9))

#nrbc
selected.nrbcalladj = matrix(0, nrow = 195, ncol = 195)
row.names(selected.nrbcalladj) = colnames(genusdata.nozeros)
colnames(selected.nrbcalladj) = colnames(genusdata.nozeros)
genusnrbclistadj = list()

for(i in 1:195){
  print(i)
  load(paste("/Volumes/rc/genusjobs/nrbcresultsadj/tdfnrbcadj", as.character(i), ".RData", sep = ""))
  genusnrbclistadj[[i]] = tdf
  match.select = match(row.names(tdf), colnames(genusdata.nozeros))
  selected.nrbcalladj[i,match.select] = selected.nrbcalladj[i,match.select] + 1
}

#Each row is sample

which(colSums(selected.nrbcalladj) > (195*.7))
which(colSums(selected.nrbcalladj) > (195*.8))
which(colSums(selected.nrbcalladj) > (195*.9))
```


Visualizing Adjusted at 90% concurrancy 
```{r}
library(UpSetR)

bcellseltaxgenadj90 = colnames(genusdata.nozeros)[which(colSums(selected.bcellalladj) > (195*.9))]
cd4tseltaxgenadj90 = colnames(genusdata.nozeros)[which(colSums(selected.cd4talladj) > (195*.9))]
cd8tseltaxgenadj90 = colnames(genusdata.nozeros)[which(colSums(selected.cd8talladj) > (195*.9))]
granseltaxgenadj90 = colnames(genusdata.nozeros)[which(colSums(selected.granalladj) > (195*.9))]
monoseltaxgenadj90 = colnames(genusdata.nozeros)[which(colSums(selected.monoalladj) > (195*.9))]
nkseltaxgenadj90 = colnames(genusdata.nozeros)[which(colSums(selected.nkalladj) > (195*.9))]
nrbcseltaxgenadj90 = colnames(genusdata.nozeros)[which(colSums(selected.nrbcalladj) > (195*.9))]


listInput90 = list(bcell = bcellseltaxgenadj90, 
                cd4t = cd4tseltaxgenadj90,
                cd8t = cd8tseltaxgenadj90,
                nk = nkseltaxgenadj90,
                mono = monoseltaxgenadj90, 
                gran = granseltaxgenadj90,
                nrbc = nrbcseltaxgenadj90)

upset(fromList(listInput90), order.by = "freq", nsets = 7, empty.intersections = "on")


unique(unlist(listInput90))
length(unique(unlist(listInput90)))


immune.crude.resultsadj90 = matrix(0, nrow = 7, ncol = length(unique(unlist(listInput90))))
colnames(immune.crude.resultsadj90) = unique(unlist(listInput90))
row.names(immune.crude.resultsadj90) = c("bcell", "cd4t", "cd8t", "gran", "mono", "nk", "nrbc")


immune.crude.resultsadj90[1,] = as.numeric(unique(unlist(listInput90)) %in% bcellseltaxgenadj90)
immune.crude.resultsadj90[2,] = as.numeric(unique(unlist(listInput90)) %in% cd4tseltaxgenadj90)
immune.crude.resultsadj90[3,] = as.numeric(unique(unlist(listInput90)) %in% cd8tseltaxgenadj90)
immune.crude.resultsadj90[4,] = as.numeric(unique(unlist(listInput90)) %in% granseltaxgenadj90)
immune.crude.resultsadj90[5,] = as.numeric(unique(unlist(listInput90)) %in% monoseltaxgenadj90)
immune.crude.resultsadj90[6,] = as.numeric(unique(unlist(listInput90)) %in% nkseltaxgenadj90)
immune.crude.resultsadj90[7,] = as.numeric(unique(unlist(listInput90)) %in% nrbcseltaxgenadj90)

immune.crude.resultsadj90
upset(fromList(listInput90), order.by = "freq", nsets = 7)

```




Visualizing Adjusted at 80% concurrancy 
```{r}
library(UpSetR)

bcellseltaxgenadj80 = colnames(genusdata.nozeros)[which(colSums(selected.bcellalladj) > (195*.8))]
cd4tseltaxgenadj80 = colnames(genusdata.nozeros)[which(colSums(selected.cd4talladj) > (195*.8))]
cd8tseltaxgenadj80 = colnames(genusdata.nozeros)[which(colSums(selected.cd8talladj) > (195*.8))]
granseltaxgenadj80 = colnames(genusdata.nozeros)[which(colSums(selected.granalladj) > (195*.8))]
monoseltaxgenadj80 = colnames(genusdata.nozeros)[which(colSums(selected.monoalladj) > (195*.8))]
nkseltaxgenadj80 = colnames(genusdata.nozeros)[which(colSums(selected.nkalladj) > (195*.8))]
nrbcseltaxgenadj80 = colnames(genusdata.nozeros)[which(colSums(selected.nrbcalladj) > (195*.8))]


listInput80 = list(bcell = bcellseltaxgenadj80, 
                cd4t = cd4tseltaxgenadj80,
                cd8t = cd8tseltaxgenadj80,
                nk = nkseltaxgenadj80,
                mono = monoseltaxgenadj80, 
                gran = granseltaxgenadj80,
                nrbc = nrbcseltaxgenadj80)

upset(fromList(listInput80), order.by = "freq", nsets = 7, empty.intersections = "on")


unique(unlist(listInput80))
length(unique(unlist(listInput80)))


immune.crude.resultsadj80 = matrix(0, nrow = 7, ncol = length(unique(unlist(listInput80))))
colnames(immune.crude.resultsadj80) = unique(unlist(listInput80))
row.names(immune.crude.resultsadj80) = c("bcell", "cd4t", "cd8t", "gran", "mono", "nk", "nrbc")


immune.crude.resultsadj80[1,] = as.numeric(unique(unlist(listInput80)) %in% bcellseltaxgenadj80)
immune.crude.resultsadj80[2,] = as.numeric(unique(unlist(listInput80)) %in% cd4tseltaxgenadj80)
immune.crude.resultsadj80[3,] = as.numeric(unique(unlist(listInput80)) %in% cd8tseltaxgenadj80)
immune.crude.resultsadj80[4,] = as.numeric(unique(unlist(listInput80)) %in% granseltaxgenadj80)
immune.crude.resultsadj80[5,] = as.numeric(unique(unlist(listInput80)) %in% monoseltaxgenadj80)
immune.crude.resultsadj80[6,] = as.numeric(unique(unlist(listInput80)) %in% nkseltaxgenadj80)
immune.crude.resultsadj80[7,] = as.numeric(unique(unlist(listInput80)) %in% nrbcseltaxgenadj80)

immune.crude.resultsadj80
upset(fromList(listInput80), order.by = "freq", nsets = 7)

```

#Regression 90%
```{r}

get.lm.func = function(data, xvar, yvar, indices){
  bootdata = data[indices,]
  lmstrep = lm(bootdata[,yvar] ~ bootdata[,xvar] + bootdata$bbymale + bootdata$enrollment_age + bootdata$imrbthwghtg_all + bootdata$gestage_all + bootdata$evercigpreg)
  return(lmstrep$coefficients)
}

dfreg90 = data.frame()

#cd4t
cd4tlmstrep = lm(genusdata.nozeros.prop.2$Streptococcus ~ immune.new2$cd4t_revised + immune.new2$bbymale + immune.new2$enrollment_age + immune.new2$imrbthwghtg_all + immune.new2$gestage_all + immune.new2$evercigpreg)
cd4tsumstrep = summary(cd4tlmstrep)
cd4t_strep_boot = boot(data = cbind(genusdata.nozeros.prop.2, immune.new2),
                       yvar = "Streptococcus",
                       xvar = "cd4t_revised",
                       statistic = get.lm.func,
                       R = 1000)

cd4t_strep_bootci = boot.ci(cd4t_strep_boot, type = "basic", index = 2)[4]

dfreg90 = rbind(dfreg90, c(cd4tsumstrep$coefficients[2,c(1,4)], confint(cd4tlmstrep)[2,], cd4t_strep_bootci[[1]][4:5]))


#mono
monolmstrep = lm(genusdata.nozeros.prop.2$Streptococcus ~ immune.new2$mono_revised + immune.new2$bbymale + immune.new2$enrollment_age + immune.new2$imrbthwghtg_all + immune.new2$gestage_all + immune.new2$evercigpreg)
monosumstrep = summary(monolmstrep)
mono_strep_boot = boot(data = cbind(genusdata.nozeros.prop.2, immune.new2),
                       yvar = "Streptococcus",
                       xvar = "mono_revised",
                       statistic = get.lm.func,
                       R = 1000)

mono_strep_bootci = boot.ci(mono_strep_boot, type = "basic", index = 2)[4]

dfreg90 = rbind(dfreg90, c(monosumstrep$coefficients[2,c(1,4)], confint(monolmstrep)[2,], mono_strep_bootci[[1]][4:5]))


#nrbc
nrbclmstrep = lm(genusdata.nozeros.prop.2$Streptococcus ~ immune.new2$nrbc_revised + immune.new2$bbymale + immune.new2$enrollment_age + immune.new2$imrbthwghtg_all + immune.new2$gestage_all + immune.new2$evercigpreg)
nrbcsumstrep = summary(nrbclmstrep)
nrbc_strep_boot = boot(data = cbind(genusdata.nozeros.prop.2, immune.new2),
                       yvar = "Streptococcus",
                       xvar = "nrbc_revised",
                       statistic = get.lm.func,
                       R = 1000)

nrbc_strep_bootci = boot.ci(nrbc_strep_boot, type = "basic", index = 2)[4]

dfreg90 = rbind(dfreg90, c(nrbcsumstrep$coefficients[2,c(1,4)], confint(nrbclmstrep)[2,], nrbc_strep_bootci[[1]][4:5]))

rownames(dfreg90) = c("cd4t","mono","nrbc")
colnames(dfreg90) = c("Estimate", "P-value", "Lower CI", "Upper CI", "Bootstrap Lower CI", "Bootstrap Upper CI")
kable(dfreg90)
dfreg90strep = dfreg90

#Enterococcus
dfreg90entero = data.frame()
#mono
monolmentero = lm(genusdata.nozeros.prop.2$Enterococcus ~ immune.new2$mono_revised + immune.new2$bbymale + immune.new2$enrollment_age + immune.new2$imrbthwghtg_all + immune.new2$gestage_all + immune.new2$evercigpreg)
monosumentero = summary(monolmentero)
mono_entero_boot = boot(data = cbind(genusdata.nozeros.prop.2,
                                     immune.new2),
                       yvar = "Enterococcus",
                       xvar = "mono_revised",
                       statistic = get.lm.func,
                       R = 1000)

mono_entero_bootci = boot.ci(mono_entero_boot, type = "basic", index = 2)[4]

dfreg90entero = rbind(dfreg90entero, c(monosumentero$coefficients[2,c(1,4)], confint(monolmentero)[2,], mono_entero_bootci[[1]][4:5]))

rownames(dfreg90entero) = c("mono")
colnames(dfreg90entero) = c("Estimate", "P-value", "Lower CI", "Upper CI", "Bootstrap Lower CI", "Bootstrap Upper CI")
kable(dfreg90entero)
```

#Regression: 80%
```{r}
#Clostridium_sensu_stricto, Enterococcus, Streptococcus, Escherichia/Shigella

dfreg80 = data.frame()

#Streptococcus
#gran
granlmstrep = lm(genusdata.nozeros.prop.2$Streptococcus ~ immune.new2$gran_revised + immune.new2$bbymale + immune.new2$enrollment_age + immune.new2$imrbthwghtg_all + immune.new2$gestage_all + immune.new2$evercigpreg)
gransumstrep = summary(granlmstrep)
gran_strep_boot = boot(data = cbind(genusdata.nozeros.prop.2, immune.new2),
                       yvar = "Streptococcus",
                       xvar = "gran_revised",
                       statistic = get.lm.func,
                       R = 1000)

gran_strep_bootci = boot.ci(gran_strep_boot, type = "basic", index = 2)[4]

dfreg80 = rbind(dfreg90, c(gransumstrep$coefficients[2,c(1,4)], confint(granlmstrep)[2,], gran_strep_bootci[[1]][4:5]))

#nk
nklmstrep = lm(genusdata.nozeros.prop.2$Streptococcus ~ immune.new2$nk_revised + immune.new2$bbymale + immune.new2$enrollment_age + immune.new2$imrbthwghtg_all + immune.new2$gestage_all + immune.new2$evercigpreg)
nksumstrep = summary(nklmstrep)
nk_strep_boot = boot(data = cbind(genusdata.nozeros.prop.2, immune.new2),
                       yvar = "Streptococcus",
                       xvar = "nk_revised",
                       statistic = get.lm.func,
                       R = 1000)

nk_strep_bootci = boot.ci(nk_strep_boot, type = "basic", index = 2)[4]

dfreg80 = rbind(dfreg80, c(nksumstrep$coefficients[2,c(1,4)], confint(nklmstrep)[2,], nk_strep_bootci[[1]][4:5]))
dfreg80strep = dfreg80


rownames(dfreg80strep) = c("cd4t","mono","nrbc", "gran", "nk")
dfreg80strep = dfreg80strep[c(order(rownames(dfreg80strep))),]
colnames(dfreg80strep) = c("Estimate", "P-value", "Lower CI", "Upper CI", "Bootstrap Lower CI", "Bootstrap Upper CI")

kable(dfreg80strep)



#Enterococcus
dfreg80entero = data.frame()

#bcell
bcelllmentero = lm(genusdata.nozeros.prop.2$Enterococcus ~ immune.new2$bcell_revised + immune.new2$bbymale + immune.new2$enrollment_age + immune.new2$imrbthwghtg_all + immune.new2$gestage_all + immune.new2$evercigpreg)
bcellsumentero = summary(bcelllmentero)
bcell_entero_boot = boot(data = cbind(genusdata.nozeros.prop.2,
                                     immune.new2),
                       yvar = "Enterococcus",
                       xvar = "bcell_revised",
                       statistic = get.lm.func,
                       R = 1000)

bcell_entero_bootci = boot.ci(bcell_entero_boot, type = "basic", index = 2)[4]

dfreg80entero = rbind(dfreg90entero, c(bcellsumentero$coefficients[2,c(1,4)], confint(bcelllmentero)[2,], bcell_entero_bootci[[1]][4:5]))

#cd4t
cd4tlmentero = lm(genusdata.nozeros.prop.2$Enterococcus ~ immune.new2$cd4t_revised + immune.new2$bbymale + immune.new2$enrollment_age + immune.new2$imrbthwghtg_all + immune.new2$gestage_all + immune.new2$evercigpreg)
cd4tsumentero = summary(cd4tlmentero)
cd4t_entero_boot = boot(data = cbind(genusdata.nozeros.prop.2,
                                     immune.new2),
                       yvar = "Enterococcus",
                       xvar = "cd4t_revised",
                       statistic = get.lm.func,
                       R = 1000)

cd4t_entero_bootci = boot.ci(cd4t_entero_boot, type = "basic", index = 2)[4]

dfreg80entero = rbind(dfreg80entero, c(cd4tsumentero$coefficients[2,c(1,4)], confint(cd4tlmentero)[2,], cd4t_entero_bootci[[1]][4:5]))


#cd8t
cd8tlmentero = lm(genusdata.nozeros.prop.2$Enterococcus ~ immune.new2$cd8t_revised + immune.new2$bbymale + immune.new2$enrollment_age + immune.new2$imrbthwghtg_all + immune.new2$gestage_all + immune.new2$evercigpreg)
cd8tsumentero = summary(cd8tlmentero)
cd8t_entero_boot = boot(data = cbind(genusdata.nozeros.prop.2,
                                     immune.new2),
                       yvar = "Enterococcus",
                       xvar = "cd8t_revised",
                       statistic = get.lm.func,
                       R = 1000)

cd8t_entero_bootci = boot.ci(cd8t_entero_boot, type = "basic", index = 2)[4]

dfreg80entero = rbind(dfreg80entero, c(cd8tsumentero$coefficients[2,c(1,4)], confint(cd8tlmentero)[2,], cd8t_entero_bootci[[1]][4:5]))


#gran
granlmentero = lm(genusdata.nozeros.prop.2$Enterococcus ~ immune.new2$gran_revised + immune.new2$bbymale + immune.new2$enrollment_age + immune.new2$imrbthwghtg_all + immune.new2$gestage_all + immune.new2$evercigpreg)
gransumentero = summary(granlmentero)
gran_entero_boot = boot(data = cbind(genusdata.nozeros.prop.2,
                                     immune.new2),
                       yvar = "Enterococcus",
                       xvar = "gran_revised",
                       statistic = get.lm.func,
                       R = 1000)

gran_entero_bootci = boot.ci(gran_entero_boot, type = "basic", index = 2)[4]

dfreg80entero = rbind(dfreg80entero, c(gransumentero$coefficients[2,c(1,4)], confint(granlmentero)[2,], gran_entero_bootci[[1]][4:5]))


#nrbc
nrbclmentero = lm(genusdata.nozeros.prop.2$Enterococcus ~ immune.new2$nrbc_revised + immune.new2$bbymale + immune.new2$enrollment_age + immune.new2$imrbthwghtg_all + immune.new2$gestage_all + immune.new2$evercigpreg)
nrbcsumentero = summary(nrbclmentero)
nrbc_entero_boot = boot(data = cbind(genusdata.nozeros.prop.2,
                                     immune.new2),
                       yvar = "Enterococcus",
                       xvar = "nrbc_revised",
                       statistic = get.lm.func,
                       R = 1000)

nrbc_entero_bootci = boot.ci(nrbc_entero_boot, type = "basic", index = 2)[4]

dfreg80entero = rbind(dfreg80entero, c(nrbcsumentero$coefficients[2,c(1,4)], confint(nrbclmentero)[2,], nrbc_entero_bootci[[1]][4:5]))

#Concat
rownames(dfreg80entero) = c("mono", "bcell", "cd4t", "cd8t", "gran", "nrbc")
dfreg80entero = dfreg80entero[c(order(rownames(dfreg80entero))),]
colnames(dfreg80entero) = c("Estimate", "P-value", "Lower CI", "Upper CI", "Bootstrap Lower CI", "Bootstrap Upper CI")
kable(dfreg80entero)



#Clostridium_sensu_stricto
dfreg80clos = data.frame()

#bcell
bcelllmclos = lm(genusdata.nozeros.prop.2$Clostridium_sensu_stricto ~ immune.new2$bcell_revised + immune.new2$bbymale + immune.new2$enrollment_age + immune.new2$imrbthwghtg_all + immune.new2$gestage_all + immune.new2$evercigpreg)
bcellsumclos = summary(bcelllmclos)
bcell_clos_boot = boot(data = cbind(genusdata.nozeros.prop.2,
                                     immune.new2),
                       yvar = "Clostridium_sensu_stricto",
                       xvar = "bcell_revised",
                       statistic = get.lm.func,
                       R = 1000)

bcell_clos_bootci = boot.ci(bcell_clos_boot, type = "basic", index = 2)[4]

dfreg80clos = rbind(dfreg80clos, c(bcellsumclos$coefficients[2,c(1,4)], confint(bcelllmclos)[2,], bcell_clos_bootci[[1]][4:5]))


#cd4t
cd4tlmclos = lm(genusdata.nozeros.prop.2$Clostridium_sensu_stricto ~ immune.new2$cd4t_revised + immune.new2$bbymale + immune.new2$enrollment_age + immune.new2$imrbthwghtg_all + immune.new2$gestage_all + immune.new2$evercigpreg)
cd4tsumclos = summary(cd4tlmclos)
cd4t_clos_boot = boot(data = cbind(genusdata.nozeros.prop.2,
                                     immune.new2),
                       yvar = "Clostridium_sensu_stricto",
                       xvar = "cd4t_revised",
                       statistic = get.lm.func,
                       R = 1000)

cd4t_clos_bootci = boot.ci(cd4t_clos_boot, type = "basic", index = 2)[4]

dfreg80clos = rbind(dfreg80clos, c(cd4tsumclos$coefficients[2,c(1,4)], confint(cd4tlmclos)[2,], cd4t_clos_bootci[[1]][4:5]))

#cd8t
cd8tlmclos = lm(genusdata.nozeros.prop.2$Clostridium_sensu_stricto ~ immune.new2$cd8t_revised + immune.new2$bbymale + immune.new2$enrollment_age + immune.new2$imrbthwghtg_all + immune.new2$gestage_all + immune.new2$evercigpreg)
cd8tsumclos = summary(cd8tlmclos)
cd8t_clos_boot = boot(data = cbind(genusdata.nozeros.prop.2,
                                     immune.new2),
                       yvar = "Clostridium_sensu_stricto",
                       xvar = "cd8t_revised",
                       statistic = get.lm.func,
                       R = 1000)

cd8t_clos_bootci = boot.ci(cd8t_clos_boot, type = "basic", index = 2)[4]

dfreg80clos = rbind(dfreg80clos, c(cd8tsumclos$coefficients[2,c(1,4)], confint(cd8tlmclos)[2,], cd8t_clos_bootci[[1]][4:5]))

#mono
monolmclos = lm(genusdata.nozeros.prop.2$Clostridium_sensu_stricto ~ immune.new2$mono_revised + immune.new2$bbymale + immune.new2$enrollment_age + immune.new2$imrbthwghtg_all + immune.new2$gestage_all + immune.new2$evercigpreg)
monosumclos = summary(monolmclos)
mono_clos_boot = boot(data = cbind(genusdata.nozeros.prop.2,
                                     immune.new2),
                       yvar = "Clostridium_sensu_stricto",
                       xvar = "mono_revised",
                       statistic = get.lm.func,
                       R = 1000)

mono_clos_bootci = boot.ci(mono_clos_boot, type = "basic", index = 2)[4]

dfreg80clos = rbind(dfreg80clos, c(monosumclos$coefficients[2,c(1,4)], confint(monolmclos)[2,], mono_clos_bootci[[1]][4:5]))

#nrbc
nrbclmclos = lm(genusdata.nozeros.prop.2$Clostridium_sensu_stricto ~ immune.new2$nrbc_revised + immune.new2$bbymale + immune.new2$enrollment_age + immune.new2$imrbthwghtg_all + immune.new2$gestage_all + immune.new2$evercigpreg)
nrbcsumclos = summary(nrbclmclos)
nrbc_clos_boot = boot(data = cbind(genusdata.nozeros.prop.2,
                                     immune.new2),
                       yvar = "Clostridium_sensu_stricto",
                       xvar = "nrbc_revised",
                       statistic = get.lm.func,
                       R = 1000)

nrbc_clos_bootci = boot.ci(nrbc_clos_boot, type = "basic", index = 2)[4]

dfreg80clos = rbind(dfreg80clos, c(nrbcsumclos$coefficients[2,c(1,4)], confint(nrbclmclos)[2,], nrbc_clos_bootci[[1]][4:5]))

#Concat
rownames(dfreg80clos) = c("bcell", "cd4t", "cd8t", "mono", "nrbc")
dfreg80clos = dfreg80clos[c(order(rownames(dfreg80clos))),]
colnames(dfreg80clos) = c("Estimate", "P-value", "Lower CI", "Upper CI", "Bootstrap Lower CI", "Bootstrap Upper CI")
kable(dfreg80clos)


#Escherichia/Shigella
dfreg80escher = data.frame()

#cd8t
cd8tlmescher = lm(genusdata.nozeros.prop.2[,"Escherichia/Shigella"] ~ immune.new2$cd8t_revised + immune.new2$bbymale + immune.new2$enrollment_age + immune.new2$imrbthwghtg_all + immune.new2$gestage_all + immune.new2$evercigpreg)
cd8tsumescher = summary(cd8tlmescher)
cd8t_escher_boot = boot(data = cbind(genusdata.nozeros.prop.2,
                                     immune.new2),
                       yvar = "Escherichia/Shigella",
                       xvar = "cd8t_revised",
                       statistic = get.lm.func,
                       R = 1000)

cd8t_escher_bootci = boot.ci(cd8t_escher_boot, type = "basic", index = 2)[4]

dfreg80escher = rbind(dfreg80escher, c(cd8tsumescher$coefficients[2,c(1,4)], confint(cd8tlmescher)[2,], cd8t_escher_bootci[[1]][4:5]))

#nk
nklmescher = lm(genusdata.nozeros.prop.2[,"Escherichia/Shigella"] ~ immune.new2$nk_revised + immune.new2$bbymale + immune.new2$enrollment_age + immune.new2$imrbthwghtg_all + immune.new2$gestage_all + immune.new2$evercigpreg)
nksumescher = summary(nklmescher)
nk_escher_boot = boot(data = cbind(genusdata.nozeros.prop.2,
                                     immune.new2),
                       yvar = "Escherichia/Shigella",
                       xvar = "nk_revised",
                       statistic = get.lm.func,
                       R = 1000)

nk_escher_bootci = boot.ci(nk_escher_boot, type = "basic", index = 2)[4]

dfreg80escher = rbind(dfreg80escher, c(nksumescher$coefficients[2,c(1,4)], confint(nklmescher)[2,], nk_escher_bootci[[1]][4:5]))


#nrbc
nrbclmescher = lm(genusdata.nozeros.prop.2[,"Escherichia/Shigella"] ~ immune.new2$nrbc_revised + immune.new2$bbymale + immune.new2$enrollment_age + immune.new2$imrbthwghtg_all + immune.new2$gestage_all + immune.new2$evercigpreg)
nrbcsumescher = summary(nrbclmescher)
nrbc_escher_boot = boot(data = cbind(genusdata.nozeros.prop.2,
                                     immune.new2),
                       yvar = "Escherichia/Shigella",
                       xvar = "nrbc_revised",
                       statistic = get.lm.func,
                       R = 1000)

nrbc_escher_bootci = boot.ci(nrbc_escher_boot, type = "basic", index = 2)[4]

dfreg80escher = rbind(dfreg80escher, c(nrbcsumescher$coefficients[2,c(1,4)], confint(nrbclmescher)[2,], nrbc_escher_bootci[[1]][4:5]))



#Concat
rownames(dfreg80escher) = c("cd8t", "nk", "nrbc")
dfreg80escher = dfreg80escher[c(order(rownames(dfreg80escher))),]
colnames(dfreg80escher) = c("Estimate", "P-value", "Lower CI", "Upper CI", "Bootstrap Lower CI", "Bootstrap Upper CI")
kable(dfreg80escher)
```


#Saving
```{r}
save(immune.new2, immune.crude.resultsadj80, listInput80, listInput90, immune.crude.resultsadj90, genusdata.nozeros.prop.2, genusdata.nozeros, dfreg90entero, dfreg90strep, dfreg90strep,dfreg80strep, dfreg80entero,dfreg80escher, dfreg80clos, 
     file = "/Volumes/rc/Dissertation Materials/CordProject/MicroImmune/FinalResults/FinalResults.RData")
```

#Using Quang Package
```{r}
library(devtools)
install_github("quangnguyen1995/MZILN")
library(MZILN)
library(utils)


set.seed(123)
system.time(
  try(
  quang.results = MZILN.main(df.main = newgenusdata,
                             df.covar = df.cd4tadj, 
                             covariates = colnames(df.cd4tadj),
                             n.lam = 200, lam.min.ratio = 1e-04, 
                             reg.method = "mcp", n.folds = 5)
)

```



#Metagenomics
```{r}
match(mbl_id_6w_humann2, gsub(" ", "", immune.new$mblid, fixed = TRUE))
```



```{r}
stool_breastmilk_16s = read.csv("/afs/northstar.dartmouth.edu/users/m/margk.collab/AnneHoen/Lab/DupeCoker/GroupFiles/SourceFiles/NHBCSreq03Jun2020_MCocker/stool_breastmilk_16s.csv")

rownames(asvdata2)

stool_breastmilk_16s_73 = stool_breastmilk_16s[na.omit(match(rownames(asvdata2), stool_breastmilk_16s$MBL_ID)),]
stool_breastmilk_16s_73$unq_id
stool_breastmilk_16s_73$MBL_ID

metadata_aim2 = read.csv("/afs/northstar.dartmouth.edu/udata/margk/collab/YukaMoroishi/SourceFiles/NHBCSreq19jul2019_YukaMoroishi/req19jul2019_ym_06jul2020.csv", na.strings=c("", "NA"))

metadata_aim2_73 = metadata_aim2[na.omit(match(stool_breastmilk_16s_73$unq_id, metadata_aim2$unq_id)),]

table(metadata_aim2_73$deliverytype)

summary(glm(immune.new2$bcell_revised ~ metadata_aim2_73$deliverytype))
summary(glm(immune.new2$mono_revised ~ metadata_aim2_73$deliverytype))
summary(glm(immune.new2$cd4t_revised ~ factor(metadata_aim2_73$deliverytype))) #Sig
summary(glm(immune.new2$cd8t_revised ~ metadata_aim2_73$deliverytype))
summary(glm(immune.new2$gran_revised ~ metadata_aim2_73$deliverytype))
summary(glm(immune.new2$nk_revised ~ metadata_aim2_73$deliverytype))
summary(glm(immune.new2$nrbc_revised ~ metadata_aim2_73$deliverytype))
```

